### PR TITLE
refactoring prometheus module and add thanos

### PR DESCRIPTION
**Pull request by akastav**
* update Prometheus, Grafana
* use separate charts
* add Thanos with s3 or Minio backend

### CODE REVIEW

**Comment by akastav** [comment]
#138

File: README.md [hunk 1]
```diff
@@ -1,16 +1,53 @@
...
+## Thanos
+Install the [thanos](https://github.com/bitnami/charts/tree/master/bitnami/thanos), long term storage capabilities for prometheus
 ## Example
 ``` hcl
 module "prometheus" {
-  source        = "git::https://github.com/provectus/sak-prometheus.git"
-  cluster_name      = module.kubernetes.cluster_name
-  argocd            = module.argocd.state
-  domains           = local.domain
+  depends_on      = [module.argocd]
```
Comment (RustamGimadiev) [inline README.md:20]: is it require? you already use outputs of argocd module as inputs for this module

File: main.tf [hunk 2]
```diff
@@ -4,10 +4,54 @@ data "aws_eks_cluster" "this" {
...
+          Action = [
+            "s3:ListBucket",
+            "s3:GetObject",
+            "s3:DeleteObject",
+            "s3:PutObject",
+            "s3:CreateBucket",
+            "s3:DeleteBucket"
+          ],
+          Resource = [
+            "arn:aws:s3:::*/*", #TODO: Change to ${aws_s3_bucket.thanos[0].id} or hardcoded name
```
Comment (RustamGimadiev) [inline main.tf]: this policy has a too wide set of permissions for that service, can we point to a certain S3 bucket with a path?

File: main.tf [hunk 3]
```diff
@@ -4,10 +4,54 @@ data "aws_eks_cluster" "this" {
 
 data "aws_region" "current" {}
 
+locals {
+  argocd_enabled     = length(var.argocd) > 0 ? 1 : 0
+  grafana_enabled    = var.grafana_enabled ? 1 : 0
+  prometheus_enabled = var.prometheus_enabled ? 1 : 0
+  thanos_enabled     = var.thanos_enabled ? 1 : 0
+  storage            = var.thanos_storage == "s3" ? 0 : 1
```
Comment (RustamGimadiev) [inline main.tf:12]: is that mean that source-specific resources will be created independently of Thanos installation? I don't see any logic between thanos_enabled and thanos_storage?  looks like s3 bucket will be created always by default

Comment (akastav) [reply main.tf:12]: it's for choose minio or s3 backend and resource "kubernetes_secret" "s3_objstore"  or resource "kubernetes_secret" "thanos_objstore"  will be created

File: README.md (hunk 1 above)
Comment (akastav) [reply README.md:20]: is it works when we used destroy command? If yes - I can delete this depend_on

**Review by RustamGimadiev — APPROVED** [review]

File: main.tf [hunk 4]
```diff
@@ -108,7 +108,7 @@ resource "aws_kms_ciphertext" "grafana_password" {
 }
 
 resource "aws_kms_ciphertext" "thanos_password" {
-  count     = 1 - local.storage
+  count     = local.storage
```
Comment (RustamGimadiev) [inline main.tf:111]: count based on string object? is it works?

Comment (akastav) [reply main.tf:111]: local.storage it's not the same that storage =)

**Pull request merged**